Automattic/jetpack
# My Jetpack: add A/A experiment

See related issue for more context

## Proposed changes:


**Note: this change has no visual effect**

* This PR adds a simple A/A experiment to My Jetpack. The two variations will have the same experience, and we are simply testing for correct random assignment and no bias in metric performance.
* It also improves the ExPlat package to not fetch the assignment at all if `anon_id` is not present. `anon_id` will only be present if Tracks is enabled. This change is mostly to prevent sending that request to the Jetpack backend.

### Other information:



## Jetpack product discussion


pbxNRc-3Rd-p2

## Does this pull request change what data or activity we track or use?


Yes.  This PR implements an A/A test on My Jetpack. With this change, we'll send a request to the backend with an anonymous id to fetch which variation the user will see (treatment or control).

## Testing instructions:






* Spin up a JN site with Jetpack Beta pointing to this branch, and click to "Enable Sandbox Access". Both options appear after clicking on "See more options"
* On your new site, go to `/wp-admin/options-general.php?page=companion_settings` and point `JETPACK__SANDBOX_DOMAIN` to your WPCOM sandbox
* Open the Network tab in your browser's developer console and filter by "assignments"
* Without connecting your site or user, go to My Jetpack `/wp-admin/admin.php?page=my-jetpack`
* Confirm that you don't see any requests listed that matches the text above
* Now, connect your site or user to WPCOM and go back to My Jetpack to perform the same step
* Confirm that the response of the request is similar to:
<img width="527" alt="image" src="https://github.com/user-attachments/assets/06e47f2a-a5d1-4e62-be60-e0c02964c7e4">

* Try going the "Audience" section on 21875-explat-experiment and hover the Bookmarklet link for Treatment. Follow the instructions there to assign an anonymous user
* Refresh My Jetpack and confirm the response of the request now has `variations: {explat_test_jetpack_implementation_aa_test: "treatment"}`
